Minio/DataModel/ObjectOperationsArgs.cs

```diff
@@ -1160,82 +1338,249 @@ public override RestRequest BuildRequest(RestRequest request)
             {
                 request.AddOrUpdateParameter("x-amz-storage-class", this.StorageClass, ParameterType.HttpHeader);
             }
+            if (this.LegalHoldEnabled != null && this.LegalHoldEnabled.Value)
+            {
+                request.AddOrUpdateParameter("x-amz-object-lock-legal-hold", "ON", ParameterType.HttpHeader);
+            }
+            if (this.ObjectLockSet)
+            {
+                if (!this.RetentionUntilDate.Equals(default(DateTime)))
+                {
+                    request.AddOrUpdateParameter("x-amz-object-lock-retain-until-date", utils.To8601String(this.RetentionUntilDate), ParameterType.HttpHeader);
+                }
+                request.AddOrUpdateParameter("x-amz-object-lock-mode",
+                                            (this.ObjectLockRetentionMode == RetentionMode.GOVERNANCE)?"GOVERNANCE":"COMPLIANCE",
+                                            ParameterType.HttpHeader);
+            }
+
             return request;
         }
     }
 
-    public class NewMultipartUploadArgs: EncryptionArgs<NewMultipartUploadArgs>
+    internal class NewMultipartUploadArgs: ObjectWriteArgs<NewMultipartUploadArgs>
     {
-        public NewMultipartUploadArgs()
+        internal bool ReplaceMetadataDirective { get; set; }
+        internal bool ReplaceTagsDirective { get; set; }
+        internal string StorageClass { get; set; }
+        internal ObjectStat SourceObjectInfo { get; set; }
+        internal CopySourceObjectArgs SourceObject { get; set; }
+        internal RetentionMode ObjectLockRetentionMode { get; set; }
+        internal DateTime RetentionUntilDate { get; set; }
+        internal bool ObjectLockSet { get; set; }
```

Comment:
why would NewMultipartUpload args reference CopySourceObjectArgs, ReplaceMetadataDirective etc - this is part of CopyObjectArgs/MultipartCopyUploadArgs isn't it?